src/Media/ImageUploaderService.php

```diff
@@ -28,17 +27,21 @@ class ImageUploaderService implements ImageUploaderInterface
 
     /**
      *  The maximum file size in bytes.
-     *  There is no limit when the file size if null.
+     *  There is no limit when the file size is null.
      */
-    private ?Natural $maxFileSize;
+    private ?int $maxFileSize;
 
     public function __construct(
         UuidGeneratorInterface $uuidGenerator,
         CommandBus $commandBus,
         FilesystemOperator $filesystem,
         string $uploadDirectory,
-        Natural $maxFileSize = null
+        int $maxFileSize = null
     ) {
+        if ($maxFileSize < 0) {
```

Comment:
There are 4 constructors in this PR. Is there any reason why you only check here if it is a positive integer?

Reply:
Good remark! I made this a bit stricter and added some extra tests for a few cases.